Library/Homebrew/command.rb

```diff
@@ -0,0 +1,146 @@
+module Homebrew
+  class Command
+    class << self
+      attr_reader :command_name, :valid_options, :description, :help_output, :man_output, :argv
+    end
+
+    def self.initialize
+      @valid_options = []
+      @description = nil
+      @command_name = nil
+      @help_output = nil
+      @man_output = nil
+      @root_options = []
+      @argv = ARGV
```

Comment:
As mentioned above: I don't think just copying this object (without even a `.dup`) is the best approach for having a pleasant interface.